bin/destroy_all.sh

```diff
@@ -1,19 +1,27 @@
 #!/bin/bash
 
-cd "${0%/*}"/.. || exit 1
+# Destroy all examples:
+# - If run without arguments, will destroy all examples from this repo
+# - If run with a PATH, will destroy examples from a given dir (recursively)
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+if [ -z "$1" ]
+then
+    cd "${0%/*}"/..
+else
+    cd "$1"
+fi
 
 for d in $(find . | grep state$ | xargs -n1 dirname)
 do
-    cd "${d}" >/dev/null || exit 1
+    cd "${d}" >/dev/null
     if [[ $(terraform show -no-color) != "" ]]
     then
         echo "In folder: ${d}"
-        if [[ $- == *i* ]]
-        then
-          terraform destroy
-        else
-          terraform destroy -auto-approve
-        fi
+        ./destroy.sh || true
```

Comment:
@ianmiell  here I wanted to run destroy for all examples even if some destroy fails.
But not sure if it's good enough this way, or we should indicate the status of the whole operation in the end.